vscode/core/src/solutionWorkflowOrchestrator.ts

```diff
@@ -0,0 +1,556 @@
+      this.state.currentQueueManager.dispose();
+      this.state.currentQueueManager = undefined;
+    }
+
+    // Clean up pending interactions
+    if (this.state.pendingInteractionsMap && this.state.pendingInteractionsMap.size > 0) {
+      this.state.pendingInteractionsMap.clear();
+      this.state.pendingInteractionsMap = undefined;
+    }
+
+    // Clean up resolver
+    if (this.state.resolvePendingInteraction) {
+      this.state.resolvePendingInteraction = undefined;
+    }
+
+    // Clear modified files
+    this.state.modifiedFiles.clear();
+
+    // Reset state flags to prevent stale UI overlays
+    this.state.mutateData((draft) => {
+      draft.isWaitingForUserInteraction = false;
+      draft.isProcessingQueuedMessages = false;
+      draft.chatMessages = [];
+    });
+  }
+
+  /**
+   * Set up the queue manager for handling workflow messages
+   */
+  private setupQueueManager(): void {
+    this.queueManager = new MessageQueueManager(
+      this.state,
+      this.workflow,
+      this.modifiedFilesPromises,
+      this.processedTokens,
+      this.pendingInteractions,
+    );
+
+    // Store references in state for external access
+    this.state.currentQueueManager = this.queueManager;
+    this.state.pendingInteractionsMap = this.pendingInteractions;
+
+    // Set up the resolver function
+    this.state.resolvePendingInteraction = this.createInteractionResolver();
+
+    // Register onDrain handler to trigger cleanup when queue empties
+    // This catches cases where the queue drains naturally without resolver calls
+    this.queueManager.onDrain(() => {
+      this.handleQueueDrained();
+    });
+  }
+
+  /**
+   * Handle queue drain event - check if cleanup should happen
+   * Called automatically when the queue becomes empty
+   */
+  private handleQueueDrained(): void {
+    this.logger.debug("Queue drained, checking cleanup conditions", {
+      workflowRunCompleted: this.workflowRunCompleted,
+      pendingInteractionsSize: this.pendingInteractions.size,
+      isWaitingForUserInteraction: this.state.data.isWaitingForUserInteraction,
+      isFetchingSolution: this.state.data.isFetchingSolution,
+    });
+
+    // Check all cleanup conditions
+    const allComplete =
+      this.pendingInteractions.size === 0 &&
+      !this.state.data.isWaitingForUserInteraction &&
+      this.queueManager!.getQueueLength() === 0;
+
+    if (allComplete && this.workflowRunCompleted && this.state.data.isFetchingSolution) {
+      this.logger.info("Queue drained and all conditions met - triggering cleanup");
+      this.finalCleanup();
+    } else {
+      this.logger.debug("Queue drained but not all conditions met for cleanup", {
+        allComplete,
+        workflowRunCompleted: this.workflowRunCompleted,
+        isFetchingSolution: this.state.data.isFetchingSolution,
+      });
+    }
+  }
+
+  /**
+   * Create the resolver function for pending interactions
+   */
+  private createInteractionResolver() {
```

Comment:
Moved the resolver function here for when accept/reject is handled in one way or another in modified file msg flow.